### PR TITLE
feat: improve filter to match title and values

### PR DESCRIPTION
**Pull request by PVinaches**
### What changed
- **`getFilteredProperties`** now accepts an optional `model` parameter (current form data) and matches properties on three criteria: property key, schema `title`, and the current runtime value.
- Nested **object** properties are recursively filtered using the model slice for that key.
- **Array** properties with nested item schemas are surfaced when either the item schema titles match or any item's runtime values match the filter.
- Empty-filter path is short-circuited: returns the original `properties` reference unchanged when there's nothing to filter or omit (avoids unnecessary re-renders).
- **`ObjectFieldGrouping`** passes the relevant model slice down to `getFilteredProperties` and strips spaces from the query term before matching.
- Import cleanup: replaced `KaotoSchemaDefinition` with the standard `JSONSchema4` type in tests.

### Tests added
- `get-filtered-properties.test.ts`: 10 new cases covering title matching, value matching, object/array recursion, empty-filter short-circuit, and edge cases (null array items, plain-string arrays).
- `ObjectFieldGrouping.test.tsx`: 2 new cases — space-stripping in the filter and required-field indicator rendering.

https://github.com/user-attachments/assets/b6881920-1954-4729-9a4c-958570b12c59

fix: https://github.com/KaotoIO/kaoto/issues/3068

## Summary by CodeRabbit

- **New Features**
  - Enhanced form field filtering to match property keys, schema titles, and current model values.
  - Improved support for filtering within nested objects and array fields (including deeper nested structures).
  - Filters now handle extra whitespace more reliably.
  - Object-schema required properties now show a required-field indicator, while non-required properties do not.

- **Bug Fixes**
  - Fixed cases where matches for nested fields, runtime values, and primitive/array inputs were not correctly surfaced.

- **Tests**
  - Added comprehensive coverage for the new filtering behavior.

### CODE REVIEW

**Review by igarashitm — APPROVED** [review]

File: src/form/utils/SchemaPropertyFilter.ts [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+  private static hasArrayItemProperties(schema: JSONSchema4): schema is JSONSchema4 & {
+    items: JSONSchema4 & { properties: NonNullable<JSONSchema4['properties']> };
+  } {
+    return (
+      schema.type === 'array' &&
+      isDefined(schema.items) &&
+      !Array.isArray(schema.items) &&
+      isDefined((schema.items as JSONSchema4).properties)
+    );
+  }
```
[the remark this answers is not in the document]
Comment (lordrip) [reply src/form/utils/SchemaPropertyFilter.ts:109]: Without type, we cannot render the appropriate form, so I don't think we should/can cover this

**Review by lordrip — APPROVED** [review]
